doc/design/software-bridge-management.md

```diff
@@ -0,0 +1,460 @@
+}
+
+// OVSInterfaceConfig contains some options from the Interface table of the OVSDB for PF
+type OVSInterfaceConfig struct {
+	Type        string            `json:"type,omitempty"`
+	Options     map[string]string `json:"options,omitempty"`
+	ExternalIDs map[string]string `json:"external_ids,omitempty"`
+	OtherConfig map[string]string `json:"other_config,omitempty"`
+}
+
+// LinuxBrConfig optional configuration for Linux bridge and uplink interface
+type LinuxBrConfig struct {
+	Interface map[string]string    // TODO clarify required settings
+	Bridge    *LinuxBrBridgeConfig `json:"bridge,omitempty"`
+}
+
+// LinuxBrBridgeConfig contains some options for linux bridge
+type LinuxBrBridgeConfig struct {
+	VlanFiltering bool `json:"vlan_filtering,omitempty"`
+	// +kubebuilder:validation:Enum=802.1Q;802.1ad
+	VlanProtocol string `json:"vlan_protocol,omitempty"`
+}
+```
+
+Open questions:
+- do we want to support each OVS Bridge/Interface option explicitly or should we support
+ generic map[string]string struct which allows set any field in the Bridge/Interface tables?
+- need to clarify list of required args for OVS and Linux bridge
+
+_**Note**: multiple policies can match single PF (vf range use-case), bridge settings from the policy with the higher priority
+will be applied to PF._
+
+#### SriovNetworkNodeState CRD
+
+```golang
+type Interface struct {
+	PciAddress        string    `json:"pciAddress"`
+	NumVfs            int       `json:"numVfs,omitempty"`
+	Mtu               int       `json:"mtu,omitempty"`
+	Name              string    `json:"name,omitempty"`
+	LinkType          string    `json:"linkType,omitempty"`
+	EswitchMode       string    `json:"eSwitchMode,omitempty"`
+	VfGroups          []VfGroup `json:"vfGroups,omitempty"`
+	ExternallyManaged bool      `json:"externallyManaged,omitempty"`
+	Bridge            *Bridge   `json:"bridge,omitempty"`
+}
+
+
+type InterfaceExt struct {
+	Name              string            `json:"name,omitempty"`
+	Mac               string            `json:"mac,omitempty"`
+	Driver            string            `json:"driver,omitempty"`
+	PciAddress        string            `json:"pciAddress"`
+	Vendor            string            `json:"vendor,omitempty"`
+	DeviceID          string            `json:"deviceID,omitempty"`
+	NetFilter         string            `json:"netFilter,omitempty"`
+	Mtu               int               `json:"mtu,omitempty"`
+	NumVfs            int               `json:"numVfs,omitempty"`
+	LinkSpeed         string            `json:"linkSpeed,omitempty"`
+	LinkType          string            `json:"linkType,omitempty"`
+	EswitchMode       string            `json:"eSwitchMode,omitempty"`
+	ExternallyManaged bool              `json:"externallyManaged,omitempty"`
+	TotalVfs          int               `json:"totalvfs,omitempty"`
+	VFs               []VirtualFunction `json:"Vfs,omitempty"`
+	Bridge            *Bridge           `json:"bridge,omitempty"`
```

Comment:
Is this the same type as we have in the `SriovNetworkNodeState.Spec` and in the `SriovNetworkNodeState.Spec`? If this information comes from an inspection on the host, it might reflect the output of a `ovs-vsctl` or `ip link`, right?

Reply:
yes, currently I think that it is possible to use the same type here and in the policy/state.

> `ovs-vsctl` or `ip link`
just as a  note, I think we will use direct requests to ovsd  to configure ovs bridges and rely on netlink for linux bridges.
Usage of cmd tools is not expected.

Reply:
> agree we should use netlink (if we miss something we can add to the golang netlink) and for ovsdb I think we can use the same package ovn-k use?

yes, I think we have options to avoid usage of cmd tools